### PR TITLE
fallback to rest of the world for payments if its not excluded

### PR DESCRIPTION
**Pull request by andymckay**
- [x] fix tests
- [x] test it end to end
- [x] drink some tea

### CODE REVIEW

**Comment by andymckay** [comment]
With rest of world selected:
screenshot 2015-01-14 14 31 49

Without rest of world selected:
screenshot 2015-01-14 14 29 11

File: media/css/devreg/payments.styl [hunk 1]
```diff
@@ -560,6 +560,10 @@ label.disabled {
 
 .paid-regions {
     margin-bottom: 10px;
+
+    .rest-of-world {
+        display: none;
```
Comment (muffinresearch) [inline media/css/devreg/payments.styl]: What's this needed for?

**Comment by muffinresearch** [comment]
What's the easiest way to add data to test this out?

File: mkt/developers/templates/developers/payments/premium.html [hunk 2]
```diff
@@ -239,14 +239,20 @@ <h2 id="paid-tab-header">
                           {% endfor %}
                         </tbody>
                       </table>
+                      <p class="hint note rest-of-world">
+                        {% trans %}
+                          Because "Rest of World" is selected, the following
+                          selected other regions will be available for payments
+                          using the "Rest of World" value.
+                          selected other regions will be available for payments.
```
Comment (muffinresearch) [inline mkt/developers/templates/developers/payments/premium.html]: This last sentence reads a bit oddly and is missing the capital letter.

File: mkt/developers/templates/developers/payments/premium.html [hunk 3]
```diff
@@ -239,14 +239,20 @@ <h2 id="paid-tab-header">
                           {% endfor %}
                         </tbody>
                       </table>
+                      <p class="hint note rest-of-world">
+                        {% trans %}
+                          Because "Rest of World" is selected, the following
```
Comment (muffinresearch) [inline mkt/developers/templates/developers/payments/premium.html]: This we need to think how to term this -  this does avoid the need to bundle all paid regions up with the other paid regions as soon as the rest of the world checkbox is checked - which makes me think this is a great place to start.

Should this mention USD?

Here's a suggestion for an amendment:

'As "Rest of World" is selected your app will be available for purchase by credit-card in all enabled "other regions" using USD"

Other regions could link to the Other Regions heading via a fragment identifer.

File: mkt/prices/models.py [hunk 4]
```diff
@@ -133,29 +132,33 @@ def get_price_currency(self, carrier=None, region=None, provider=None):
 
         return price_currency
 
-    def get_price_data(self, carrier=None, region=None, provider=None):
+    def get_price_data(self, carrier=None, regions=None, provider=None):
         """
         Returns a tuple of Decimal(price), currency, locale.
```
Comment (muffinresearch) [inline mkt/prices/models.py]: Looks like this comment might have been out of date already.

File: mkt/webapps/tests/test_models.py [hunk 5]
```diff
@@ -902,6 +906,9 @@ def test_has_premium(self):
 
     def test_get_price_no_premium(self):
         webapp = Webapp(premium_type=mkt.ADDON_PREMIUM)
+        webapp.save()
+        # Needed because get_price access excluded, which triggers geodata
```
Comment (muffinresearch) [inline mkt/webapps/tests/test_models.py]: typo - should that be 'accesses'?

File: mkt/webapps/tests/test_models.py [hunk 6]
```diff
@@ -1456,19 +1463,49 @@ def setUp(self):
...
+    def test_premium_remove_tier(self):
+        self.make_tier()
+        self.app.addonexcludedregion.create(region=mkt.regions.RESTOFWORLD.id)
+        # If we exclude the rest of the world, then we'll exclude Nicaragua
+        # which has no price currency.
+        ok_(mkt.regions.NI.id in self.app.get_excluded_region_ids())
+
+    def test_not_paid_worldwide(self):
+        self.make_tier()
+        self.row.update(paid=False)
```
Comment (muffinresearch) [inline mkt/webapps/tests/test_models.py]: Good call.

**Comment by muffinresearch** [comment]
The interface in Compatibility & Payments looks to work as advertised with the right currencies added.

Deffo worth adding a qa+ on the bug so all the use-cases can get thoroughly tested. 

r+wc

File: mkt/developers/templates/developers/payments/premium.html (hunk 3 above)
Comment (andymckay) [inline mkt/developers/templates/developers/payments/premium.html]: I like that wording. It explicitly says USD and credit card though. So will go for:

As "Rest of World" is selected your app will be available for purchase in all enabled "Other Regions"

... and link it up.

**Pull request merged**